source/Rod.cpp

```diff
@@ -956,9 +956,15 @@ Rod::doRHS()
 		// >>> add Pd variable for dynamic pressure, which will be applied
 		// on Rod surface
 		auto surface_height = 0.5 * (zeta[i] + zeta[i + 1]);
-		
-		if (i == N) VOF0 = calcSubSeg(r[i-1], r[i], surface_height, d);
-		else VOF0 = calcSubSeg(r[i], r[i + 1], surface_height, d);
+
+		if (N == 0) {
+			VOF0 = 1.0 ? r[i].z() < surface_height : 0.0;
```

Comment:
FYI this line is incorrect, should be `VOF0 = r[i].z() < surface_height ? 1.0 : 0.0;`

Although this whole section actually is malformed, and will read off the end of the zeta array, should be 
```c++
		if (N == 0) {
			VOF0 = r[i].z() < zeta[i] ? 1.0 : 0.0;
		} else {
			if (i == N) {
				auto surface_height = 0.5 * (zeta[i - 1] + zeta[i]);
				VOF0 = calcSubSeg(r[i - 1], r[i], surface_height, d);

			} else {
				auto surface_height = 0.5 * (zeta[i] + zeta[i + 1]);
				VOF0 = calcSubSeg(r[i], r[i + 1], surface_height, d);
			}
		}
```

Reply:
Fixed on 4561d264217ba725c150f34bb4babbaa42b313db